app/src/main/java/com/github/se/cyrcle/model/review/ReviewViewModel.kt

```diff
@@ -31,21 +30,39 @@ class ReviewViewModel(
   val userReviews: StateFlow<List<Review?>> = _userReviews
 
   /** Selected parking to review/edit */
-  private val _selectedReviewReports = MutableStateFlow<List<ReviewReport>?>(null)
-  val selectedReviewReports: StateFlow<List<ReviewReport>?> = _selectedReviewReports
+  private val _selectedReviewReports = MutableStateFlow<List<ReviewReport>>(emptyList())
+  val selectedReviewReports: StateFlow<List<ReviewReport>> = _selectedReviewReports
 
   fun addReview(review: Review) {
     reviewRepository.addReview(review, {}, { Log.e("ReviewViewModel", "Error adding review", it) })
   }
 
   fun selectReview(review: Review) {
     _selectedReview.value = review
+    loadReportsForSelectedReview()
```

Comment:
selectReview is called in AdminScreen which is what it's intended for because of the loadReportsForSelectedReview().
But it's also called in the AllReviewsScreen. Im not sure why AllReviewsScreen would need to loadReportsForSelectedReview() ? Tell me if i misunderstood something, but it just seems to me like an unnecessary call, but it's not a big deal.

Reply:
So we could have like 2 different functions, one which selects and load reports for admin, and one which selects simply.